### PR TITLE
[FLINK-10384][table] Add Sinh math function supported in Table API and SQL

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request adds Sinh math function supported in Table API and SQL*

## Brief change log

  - *Adds Sinh math function supported in Table API and SQL*

## Verifying this change

This change is already covered by existing tests, such as *ScalarFunctionsTest#testSinh*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

**Comment by yanghua** [comment]
cc @xccui

**Comment by yanghua** [comment]
@pnowojski If you have time, you can also help me to review this PR. thanks.

**Comment by pnowojski** [comment]
Yes, I was planning to after merging the `cosh` PR :) Before I take a look, could you also apply the same changes that I asked for `cosh` to this one (if appropriate).

**Comment by yanghua** [comment]
OK, will update it based on the PR of `cosh`.

**Comment by yanghua** [comment]
@pnowojski also refactored this PR's documentation.

**Comment by yanghua** [comment]
also updated the documentation.

**Comment by yanghua** [comment]
@pnowojski Yes, you are right, this will greatly improve efficiency. Next time if there is a similar PR, I will pay attention, thank you.

File: docs/dev/table/functions.md [hunk 1]
```diff
@@ -1197,6 +1197,18 @@ SIN(numeric)
...
 
+    <tr>
+      <td>
+        {% highlight text %}
+SINH(numeric)
+{% endhighlight %}
+      </td>
+      <td>
+        <p>Returns the hyperbolic sine of <i>NUMERIC</i>.</p> 
+        <p>Return value type is <i>DOUBLE</i>.</p>
```
Comment (xccui) [inline docs/dev/table/functions.md]: Change `NUMERIC` to lowercase.

The return type is ...

**Comment by pnowojski** [comment]
Thanks for contribution @yanghua !

**Comment by pnowojski** [comment]
Ops. I couldn't merge this because of conflict :( @yanghua  can you rebase it? Another sign that those functions should be added in one PR :/

**Comment by yanghua** [comment]
@pnowojski  I have rebased this PR. About those functions belongs one category should put into one PR, you are right. Next time, if I also contribute built-in functions, I will follow this rule.

**Comment by pnowojski** [comment]
Thanks @yanghua :) Merging

**Pull request merged**